server/resp/src/main/java/org/infinispan/server/resp/json/JsonAppendFunction.java

```diff
@@ -0,0 +1,120 @@
+package org.infinispan.server.resp.json;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.functional.EntryView;
+import org.infinispan.functional.EntryView.ReadWriteEntryView;
+import org.infinispan.server.resp.ExternalizerIds;
+import org.infinispan.util.function.SerializableFunction;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.jayway.jsonpath.JsonPath;
+
+public class JsonAppendFunction
+        implements SerializableFunction<EntryView.ReadWriteEntryView<byte[], JsonBucket>, List<Long>> {
+    public static final AdvancedExternalizer<JsonAppendFunction> EXTERNALIZER = new JsonAppendFunction.Externalizer();
+    public AppendType appendType;
+    public byte[] path;
+    public byte[] value;
+
+    public JsonAppendFunction(byte[] path, byte[] value, AppendType appType) {
+        this.appendType = appType;
+        this.path = path;
+        this.value = value;
+    }
+
+    @Override
+    public List<Long> apply(ReadWriteEntryView<byte[], JsonBucket> entryView) {
+        var doc = entryView.find().orElse(null);
+        if (doc == null) {
+            throw new RuntimeException("could not perform this operation on a key that doesn't exist");
```

Comment:
so we launch the error here instead of returning null ? this should be done in other commands too I guess

Reply:
No, the behaviour depends by the command, i.e.:
```quote
127.0.0.1:6379> json.objlen non-existent-key .
(nil)
127.0.0.1:6379> json.strappend non-existent-key .
(error) ERR could not perform this operation on a key that doesn't exist
```